### PR TITLE
Added legal text to upgrade button, and showing again the Edition section in team edition

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Added legal text to upgrade button, and showing again the Edition section in team edition

#### Ticket Link
[MM-28811](https://mattermost.atlassian.net/browse/MM-28811)

#### Screenshots
image

### CODE REVIEW

**Review by agnivade — COMMENTED** [review]
Can we add a test?

**Comment by jespino** [comment]
Actually it is covered by the existing snapshots, I'll update them.

**Review by agnivade — APPROVED** [review]

**Comment by amyblais** [comment]
@mkraft Can you help prioritize reviewing this as this was requested for a dot release?

**Review by mkraft — APPROVED** [review]

File: components/admin_console/admin_definition.jsx [hunk 1]
```diff
@@ -199,10 +199,7 @@ const AdminDefinition = {
                 'Mattermost Enterprise Edition. Unlock enterprise features in this software through the purchase of a subscription from ',
                 'This software is offered under a commercial license.\n\nSee ENTERPRISE-EDITION-LICENSE.txt in your root install directory for details. See NOTICE.txt for information about open source software used in this system.',
             ],
-            isHidden: it.any(
-                it.not(it.enterpriseReady),
-                it.configIsTrue('ExperimentalSettings', 'RestrictSystemAdmin'),
-            ),
+            isHidden: it.configIsTrue('ExperimentalSettings', 'RestrictSystemAdmin'),
```
Comment (fmunshi) [inline components/admin_console/admin_definition.jsx:202]: There is actually a `isHidden` check above on line 187 that says 
`isHidden: it.not(it.userHasReadPermissionOnResource('about')),` 

so you could remove that line and combine it with this check here to be something like
```
isHidden: it.any(
  it.configIsTrue('ExperimentalSettings', 'RestrictSystemAdmin'),
  it.not(it.userHasReadPermissionOnResource('about')),
),
```

**Review by esadur — APPROVED** [review]
looks good to me, thanks @jespino

**Comment by esadur** [comment]
@amyblais this doesn't need to go into a dot release. It can just go into 5.28.

**Review by fmunshi — APPROVED** [review]

**Comment by amyblais** [comment]
@jespino Please help assign a QA reviewer.

**Comment by amyblais** [comment]
@ogi-m Kind reminder to review for v5.28.

**Comment by amyblais** [comment]
@ogi-m Kind reminder for v5.28.

**Review by ogi-m — APPROVED** [review]
Tested by @prapti and the text shows up as expected next to the upgrade button and in the Edition section.

**Pull request merged**